confmap/converter/expandconverter/expand.go

```diff
@@ -36,6 +38,7 @@ func newConverter(set confmap.ConverterSettings) confmap.Converter {
 	}
 }
 
+// Deprecated: [v0.105.0] BASH-style env var expansion is deprecated. Use `${FOO}` or `${env:FOO}` instead.
```

Comment:
Let's first move the `$$` escaping and ensure `${FOO}` expansion without it?

Reply:
@dmitryax I was testing something, but I believe my [original PR](https://github.com/open-telemetry/opentelemetry-collector/pull/10510) will be the one to go with